### DISCUSSION TITLE
Modifying Video Content in Real-Time on Instagram Using mitmproxy

### DISCUSSION THREAD

**Discussion by ajk-github** [mitmproxy/mitmproxy#6852] [Intercept Everything!]
Hi mitmproxy Community,

I have successfully used mitmproxy to blur images in real-time as they are served on Instagram, but I'm facing challenges when trying to apply similar modifications to video content. I'm interested in blurring videos on the Instagram feed as they are being streamed (I am eventually trying to make a tool that blurs out content on instagram that is not safe for adolescent children).

I would appreciate any guidance on the following:

Is it possible to modify video content in real-time using mitmproxy? If so, what approach would you recommend?
If real-time video modification isn't feasible with mitmproxy, are there any alternative tools or techniques that could achieve this effect?
Thank you in advance for your insights and suggestions!

**Comment by Prinzhorn** [comment]
If your goal is to protect adolescent children, why not just block the video (or serve a placeholder video)? Other than looking "cool" what purpose does blurring serve?

**Reply by ajk-github** [reply]
I am using blurring just as a proof-of-concept that we can indeed modify the videos that are being served instead of just blocking them. A later goal that is useful in the real world is giving relevant disclaimers to videos (example: putting a blood and gore warning on a video that contains blood and gore)

**Reply by Prinzhorn** [reply]
Thanks for clarifying. I don't have any specific recommendation, as this sounds like a (cpu) intensiv task. I personally wouldn't do this on-the-fly, maybe delegate this to a different server that also persists the blurred video (no reason to blur the same video every single time someone views it).

You can look into `response.stream = some_function` (https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/http-stream-modify.py) to process the video in chunks. But I don't know enough about video formats if this would be possible to blur them in arbitrary chunks.

**Reply by ajk-github** [reply]
I have been trying to use `response.stream` but the video content does not change at all,
